### ISSUE TITLE
4.1 support

### ISSUE THREAD

**Issue by MrMage750** [UuuNyaa/blender_mmd_tools#137]
the next major version of blender is here, so it's time for the next major version of mmdtools to follow

**Comment by Yusarina** [comment]
As the developer as stated in a few places now MMD Tool's won't be getting blender 4.x support unto the next LTS which will be Blender 4.2 which doesn't come out unto June/ July. This is per MMD Tool's Blender Support Policy 

![Screenshot 2024-03-30 222237](https://github.com/UuuNyaa/blender_mmd_tools/assets/104736380/3d1d3716-a058-4aed-9391-5c08e401a75f)

**Comment by MrMage750** [comment]
> As the developer as stated in a few places now MMD Tool's won't be getting full blender 4.x support unto the next LTS which will be Blender 4.2 which doesn't come out unto July. This is per MMD Tool's Blender Support Policy.
> 
> ![Screenshot 2024-03-30 222237](https://private-user-images.githubusercontent.com/104736380/318230079-3d1d3716-a058-4aed-9391-5c08e401a75f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE5MzM4OTEsIm5iZiI6MTcxMTkzMzU5MSwicGF0aCI6Ii8xMDQ3MzYzODAvMzE4MjMwMDc5LTNkMWQzNzE2LWEwNTgtNGFlZC05MzkxLTVjMDhlNDAxYTc1Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwMVQwMTA2MzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zZTI5YTc0Y2ZlYmQ5ZTk0NDk1MDU3ZWExYjVhMTMyNWI1YjkyNDI4YjY2YzY1MTJiNzhkNWMwNGFhOWY4NTBiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.6lV9pJmPBUkbqfJvbQMpasrdT5V9JiD2R0ndHIdscMM)
> 
> ![Screenshot 2024-03-30 222409](https://private-user-images.githubusercontent.com/104736380/318230125-1b890bad-0f54-4472-9106-daafb0240947.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE5MzM4OTEsIm5iZiI6MTcxMTkzMzU5MSwicGF0aCI6Ii8xMDQ3MzYzODAvMzE4MjMwMTI1LTFiODkwYmFkLTBmNTQtNDQ3Mi05MTA2LWRhYWZiMDI0MDk0Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwMVQwMTA2MzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNTgyMjIyNjk4MmVkMGQ0MmMyMDljZGQyMTc5MTI4MjE4ZTM5NWI2MTU4OGEzNWEwMWQ1ODUxMWZhNjdiNzA0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.wloVbfN2TGxSIlIrIK00koE7q8LkjApirz1N0WcWXNI)
> 
> There is a 4.x branch [here](https://github.com/UuuNyaa/blender_mmd_tools/tree/blender-v4) which may get updates for 4.1.

the thing is the current version of the addon has no 4.1 support at all, as in it has **zero** functionality. it can't even be installed, so if the dev wants to retain users, mmdtools will need enough partial support to be able to perform basic functions on the latest version of blender

**Comment by Yusarina** [comment]
Again there is an 4.x branch which you can install just fine (Some functions may show errors due to blender removing use_auto_smooth, but works fine mostly).

"so if the dev wants to retain users" 

Alot of people who are used to blender know you should never update straight to a new version of blender as most plugins won't be updated (And a few i use also only support the latest LTS) and the fact most blender versions when there come out are just broken (4.0 was a buggy mess and still kinda is). 

Alot of people stick with LTS version of blender for a reason and that fact that there isn't a other plugin like MMD Tool's (To my knowledge) I highly doubt the dev would loose people over not instantly updating the plugin for a new version of blender (People do have lives you know).

If your that desperate to have MMD Tool's on the latest version you can always fork it yourself.

**Comment by MrMage750** [comment]
> Again there is an 4.x branch which you can install just fine (Some functions may show errors due to blender removing use_auto_smooth, but works fine mostly).
> that's the one I'm using, and it won't install in 4.1. try it for yourself and see
> "so if the dev wants to retain users"
> Alot of people who are used to blender know you should never update straight to a new major version of blender unless said person needs the new features as most plugins won't be updated (And a few i use also only support the latest LTS). And the fact that most blender versions when there have been released are just broken (4.0 was a buggy mess and still kinda is). 4.1 Has it's own issues right now and some bugs from 4.0 made it into 4.1 as well.
> Alot of people stick with the latest LTS version of blender for a reason (It's stable and does not have the large amount of issues 4.x has right now) and the fact is there isn't really any other plugin like MMD Tools (To my knowledge atleast). So I highly doubt the dev would loose people over this. Especially as this has kind of been the policy of MMD Tools (And for lot of other blender plugins) for a while to only fully support the latest LTS.
> However, even with all that said 4.1 support would be cool and all, but there isn't any reason to upgrade to it. 4.0 was kinda exiting (If it didn't have the bugs it had) but 4.1 isn't as much. And honestly if your just using MMD Tools for the stuff it offers you should just stick to 3.6 unto 4.2 comes out unless you REALLY need the new features of 4.x.
>  tl;dr
> If your that desperate to have MMD Tool's on the latest version you can always fork it yourself.
>typical "if you don't like _x,_, make it yourself" ad hominem
> Also in case you don't know how to install the 4.x branch you need to unzip the mmd_tools folder and zip it up into it's own zip file, then install as normal, if your just downloading the branch and trying to install it, that won't work.
I've been using the exact same method of installing the addon in 4.0, if that's the same version of the addon, then it 100% is a compatibility issue

**Comment by MrMage750** [comment]
> Also in case you don't know how to install the 4.x branch you need to unzip the mmd_tools folder and zip it up into it's own zip file, then install as normal, if your just downloading the branch and trying to install it, that won't work

oddly enough, that worked. though installing the 4.0 branch worked in blender 4.0
